File: bidding-engine/src/test/java/com/omnibus/bidding/application/BiddingRecommendationServiceTest.java
package com.omnibus.bidding.application;

import static org.assertj.core.api.Assertions.assertThat;

import com.omnibus.bidding.domain.RecommendationRequest;
import com.omnibus.bidding.domain.RecommendationResponse;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

@SpringBootTest(properties = "omnibus.kafka.enabled=false")
class BiddingRecommendationServiceTest {

    @Autowired
    private BiddingRecommendationService biddingRecommendationService;

    @Test
    void shouldRecommendOneNoTrumpForBalancedFifteenToSeventeenHand() {
        RecommendationResponse response = biddingRecommendationService.recommend(
            new RecommendationRequest("AQJ KQ2 A43 J742", "T97 A854 Q76 K98", "", "POLISH_CLUB")
        ).block();

        assertThat(response.recommendedBid()).isEqualTo("1NT");
    }

    @Test
    void shouldRecommendPassBelowOpeningThreshold() {
        RecommendationResponse response = biddingRecommendationService.recommend(
            new RecommendationRequest("QJ4 T82 A743 942", "A83 KQ54 T62 K75", "", "POLISH_CLUB")
        ).block();

        assertThat(response.recommendedBid()).isEqualTo("PASS");
    }
}

File: services/bidding-engine/src/test/kotlin/pl/writeonly/omnibus/bidding/application/BiddingRecommendationServiceTest.kt
package pl.writeonly.omnibus.bidding.application

import pl.writeonly.omnibus.bidding.domain.RecommendationRequest
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.junit.jupiter.api.Disabled

@SpringBootTest(properties = ["omnibus.kafka.enabled=false"])
@Disabled("This test is not working after refactor, needs to be fixed")
class RestBiddingServiceTest {
    @Autowired
    private lateinit var biddingRecommendationService: RestBiddingService

    @Test
    fun shouldRecommendOneNoTrumpForBalancedFifteenToSeventeenHand() {
        val response = biddingRecommendationService.recommend(
            RecommendationRequest("AQJ KQ2 A43 J742", "T97 A854 Q76 K98", "", "POLISH_CLUB"),
        )

        requireNotNull(response)
        assertThat(response.recommendedBid).isEqualTo("1NT")
    }

    @Test
    fun shouldRecommendPassBelowOpeningThreshold() {
        val response = biddingRecommendationService.recommend(
            RecommendationRequest("QJ4 T82 A743 942", "A83 KQ54 T62 K75", "", "POLISH_CLUB"),
        )

        requireNotNull(response)
        assertThat(response.recommendedBid).isEqualTo("PASS")
    }
}
